src/frozen_solvers.jl

```diff
@@ -94,7 +97,7 @@ alg_order(::CG2) = 2
 """
     CG2Cache
 
-Cache for [`CG2`](@ref).
+Mutable cache for [`CG2`](@ref).
```

Comment:
Actually when revisiting this I noticed that some that cache should be in `integrator.k` but that will only be necessary for #8 so I left it for later. I don't really understand enough about how SciML manages `integrator.k` right now.`ManifoldEulerCache` seems to use it right but for higher-order methods I'm not sure.

Reply:
Could we still (as for my comment above) put a link to where this is adopted from?

Reply:
I don't remember, I did it three years ago. IIRC there was a lot of guessing because solvers are not entirely consistent and it's hard to tell which approach is recommended.

Reply:
one of my main criticism of SciML stuff. If you can not find it, that is also fine then just close this remark/comment. We should maybe check for future solvers to add where we got inspired from for such a framework thingy

Reply:
For future solvers it'd be best to ask on Slack or discourse.

Reply:
...and then document here carefully. Sure instead of `help?>` the `Chris?>` mode 😬